azurelinuxagent/common/protocol/extensions_goal_state.py

```diff
@@ -507,22 +538,106 @@ def __init__(self, etag, json_text):
 
         try:
             self._parse_vm_settings(json_text)
+            self._do_common_validations()
         except Exception as e:
-            # TODO: Review code to determine whether we can rename ExtensionConfigError to InvalidExtensionsGoalStateError or
-            #       if we need to differentiate between errors in ExtensionsConfig and vmSettings.
-            raise ExtensionConfigError("Error parsing vmSettings (etag: {0}): {1}\n{2}".format(etag, format_exception(e), json_text))
+            raise VmSettingsError("Error parsing vmSettings (etag: {0}): {1}\n{2}".format(etag, format_exception(e), json_text))  # <==
 
     def _parse_vm_settings(self, json_text):
-        vm_settings = json.loads(json_text)
+        vm_settings = _CaseFoldedDict.from_dict(json.loads(json_text))
+        self._parse_status_upload_blob(vm_settings)
         self._parse_required_features(vm_settings)
         # TODO: Parse all atttributes
 
+    def _parse_status_upload_blob(self, vm_settings):
+        status_upload_blob = vm_settings.get("statusUploadBlob")
+        if status_upload_blob is None:
+            raise Exception("Missing statusUploadBlob")
+        self._status_upload_blob = status_upload_blob["value"]
```

Comment:
We should also check if `value` is present in the `status_upload_blob` and raise appropriate `Exception` if not (else a `KeyError` might be raised)